sdk/core/azure-core-amqp/inc/azure/core/amqp/connection.hpp

```diff
@@ -268,6 +296,35 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
     /** @brief Destroy an AMQP connection */
     ~Connection();
 
+    /** @brief Create a session on the current Connection object.
+     *
+     * An AMQP Session provides a context for sending and receiving messages. A single connection
+     * may have multiple independent sessions active simultaneously up to the negotiated maximum
+     * channel count.
+     *
+     * @param options The options to use when creating the session.
+     * @param eventHandler The event handler for the session.
+     */
+    Session CreateSession(SessionOptions const& options = {}, SessionEvents* eventHandler = nullptr)
+        const;
+
+    /** @brief Construct a new session associated with the specified connection over the specified
+     * endpoint.
+     *
+     * @param newEndpoint - AMQP Endpoint from which to create the session.
+     * @param eventHandler - Event handler for session events.
+     *
+     * @remarks Note that this function is normally only called from a application listening for
+     * incoming connections, not from an AMQP client.
+     */
+    Session CreateSession(
+        Endpoint& newEndpoint,
```

Comment:
Dos this need to be a non-const ref?

Reply:
Yes, because we alter the state of the Endpoint object in this method.